tracer/src/Datadog.Trace/ContinuousProfiler/ProfilerStatus.cs

```diff
@@ -22,7 +22,11 @@ internal class ProfilerStatus : IProfilerStatus
 
         public ProfilerStatus()
         {
-            _isProfilingEnabled = EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.ProfilingEnabled)?.ToBoolean() ?? false;
+            var processArchitecture = FrameworkDescription.Instance.ProcessArchitecture;
+            var isSupportedArch = processArchitecture == ProcessArchitecture.X64 || processArchitecture == ProcessArchitecture.X86;
```

Comment:
Do we need to worry about x86 linux here (i.e. that's not a supported arch/os combo) 🤔. What about macOS?

Reply:
good question.
Do we ship linux x86 and macOS binaries today ?
We do not build the profiler for linux x86 nor macOS. Do you think we should ?

Reply:
Updated the code to check the OS too